docs/content/sips/xxx-trigger-executors.md

```diff
@@ -0,0 +1,65 @@
+title = "SIP xxx - Trigger Executors"
+template = "main"
+
+---
+
+Summary: A new architecture for Spin trigger executors.
+
+Owner: lann.martin@fermyon.com
+
+Created: April 29, 2022
+
+<!-- Updated: April 4, 2022 -->
+
+## Background
+
+Spin currently includes built-in implementations for two trigger types: HTTP, and Redis (PubSub).
+We want to allow for additional trigger types in the future, whether they are "internal" types developed and
+supported within the Spin codebase, or "external" types developed by the broader community.
+
+While it is currently possible to build external trigger types - as in the
+[TimerTrigger example](../../../examples/spin-timer/src/main.rs) - it requires replicating much of the
```

Comment:
nit: Could you make this an absolute GitHub URL?
(when rendered in the docs website, the relative path won't work)